tests/functional/idempotency/conftest.py

```diff
@@ -2,7 +2,9 @@
 import hashlib
 import json
 from collections import namedtuple
+from dataclasses import dataclass
```

Comment:
NOTE: The testing changes were a bit more in-depth. My goal was to parametrize the `persistence_store` fixture, that way every test using that fixture would test with both simple and composite primary keys. All of the other changes were required to support that idea. Overall, I think the code is a _bit_ more complicated, b/c fixtures now return functions in some cases; however, that complication comes with the benefit of fewer code locations of hardcoding the keys.

Reply:
Let's do this in a separate PR so we can focus on the feature at hand - there are more opportunities to clean up and make it easier.

Reply:
Reverted tests for now.